### DISCUSSION TITLE
Does Autospotting support Graviton2 and Mixed Instance Policy?

### DISCUSSION THREAD

**Discussion by bilalmughal** [LeanerCloud/AutoSpotting#460] [Q&A]
I don't know if it's the right place to ask this question but I feel this feature will not only takes the "AutoSpotting" a level above. And will help many others as well.
We can have multiple launch templates to allow instances with different CPU architectures (Arm and x86) to launch in the same Auto Scaling group. 
We can achieve this  with following cloudformation template snippet

```
MixedInstancesPolicy:
        LaunchTemplate:
          LaunchTemplateSpecification:
              LaunchTemplateId: !Ref LaunchTemplate
              Version: !GetAtt "LaunchTemplate.LatestVersionNumber"
          Overrides:
            - InstanceType: t4g.small
              LaunchTemplateSpecification:
                # Override template for Arm based CPU t4g.small
                LaunchTemplateId: !Ref LaunchTemplateArm
                Version: !GetAtt "LaunchTemplateArm.LatestVersionNumber"
            - InstanceType: t3a.small
            - InstanceType: t3.small
```
Since the mixed instance policy does not support LaunchConfiguration, so we are forced to use launch template here. And I was never able to make AutoSpotting work with the launch template anyways.

To sum up everything 
- Does Autospotting support Arm Based CPUs?
- Does Autospotting fully support launch templates?
- Does Autospotting work with `MixedInstancesPolicy`?

**Comment by cristim** [comment]
Thanks for your questions, I'll try to answer them one by one:

Does Autospotting support Arm Based CPUs?
- It probably works but I haven't tested it. Let me know if you notice any issues
  
Does Autospotting fully support launch templates?
- plain Launch Templates should be supported, but I haven't tested LTs with overrides. Probably they need some code changes.

Does Autospotting work with MixedInstancesPolicy?
- No, it's intentionally avoiding to do any actions on ASGs configured with MixedInstancesPolicy in order not to interfere negatively with the Spot configuration defined on the AutoScaling group. If you're using MixedInstancesPolicy you probably  don't need AutoSpotting.
